src/client.js

```diff
@@ -206,6 +220,7 @@ const Client = class extends Base {
     let results = [];
     let error_count = 0;
     let prev_key = null, next_key = this.start_key;
+    let end_parsed = this.end_key == null ? null : ISO8601X.parse_extended(this.end_key);
```

Comment:
This should provide a friendlier message for an invalid key – but I should also parsed `start_key` to validate it as part of #42. Not a merge blocker.